server/sonar-server/src/main/java/org/sonar/server/component/es/ProjectMeasuresQuery.java

```diff
@@ -22,23 +22,41 @@
 
 import java.util.ArrayList;
 import java.util.List;
+import java.util.Optional;
+import org.sonar.api.measures.Metric;
 
+import static com.google.common.base.Preconditions.checkArgument;
 import static java.lang.String.format;
 import static java.util.Arrays.stream;
 import static java.util.Objects.requireNonNull;
+import static org.sonar.api.measures.CoreMetrics.ALERT_STATUS_KEY;
 
 public class ProjectMeasuresQuery {
-  private List<MetricCriteria> metricCriteria = new ArrayList<>();
+  private List<MetricCriterion> metricCriteria = new ArrayList<>();
+  private Optional<Metric.Level> qualityGateStatus = Optional.empty();
```

Comment:
I prefer the same pattern, but without optional. You add `requireNonNull` in `getQualityGateStatus()`